numpy/fft/fftpack_litemodule.c

```diff
@@ -166,6 +166,10 @@ fftpack_rfftf(PyObject *NPY_UNUSED(self), PyObject *args)
     PyArray_DIMS(data)[PyArray_NDIM(data) - 1] = npts/2 + 1;
     ret = (PyArrayObject *)PyArray_Zeros(PyArray_NDIM(data),
             PyArray_DIMS(data), PyArray_DescrFromType(NPY_CDOUBLE), 0);
+    if (ret == NULL) {
+        PyErr_NoMemory();
```

Comment:
yes PyArray_Zeros is a PyObject returning function, convention is that when these return NULL an exception is already set and just needs to be bubbled up to python, in this case it is raised all the way down in `PyArray_NewFromDescr_int` which does the main array creation and memory allocation